src/plugins/expressions/common/expression_functions/specs/overall_metric.ts

```diff
@@ -0,0 +1,157 @@
+ *   overall metric of the current series - if this results in `NaN` (like in case of objects), all cells of the
+ *   current series will be set to `NaN`.
+ * * To determine separate series defined by the `by` columns, the values of these columns will be cast to strings
+ *   before comparison. If the values are objects, the return value of their `toString` method will be used for comparison.
+ *   Missing values (`null` and `undefined`) will be treated as empty strings.
+ */
+export const overallMetric: ExpressionFunctionOverallMetric = {
+  name: 'overall_metric',
+  type: 'datatable',
+
+  inputTypes: ['datatable'],
+
+  help: i18n.translate('expressions.functions.overallSum.help', {
+    defaultMessage: 'Calculates the overall sum of a column in a data table',
+  }),
+
+  args: {
+    by: {
+      help: i18n.translate('expressions.functions.overallSum.args.byHelpText', {
+        defaultMessage: 'Column to split the overall sum calculation by',
+      }),
+      multi: true,
+      types: ['string'],
+      required: false,
+    },
+    metric: {
+      help: i18n.translate('expressions.functions.overallSum.metricHelpText', {
+        defaultMessage: 'Metric to calculate',
+      }),
+      types: ['string'],
+      options: ['sum', 'min', 'max', 'average'],
+    },
+    inputColumnId: {
+      help: i18n.translate('expressions.functions.overallSum.args.inputColumnIdHelpText', {
+        defaultMessage: 'Column to calculate the overall sum of',
+      }),
+      types: ['string'],
+      required: true,
+    },
+    outputColumnId: {
+      help: i18n.translate('expressions.functions.overallSum.args.outputColumnIdHelpText', {
+        defaultMessage: 'Column to store the resulting overall sum in',
+      }),
+      types: ['string'],
+      required: true,
+    },
+    outputColumnName: {
+      help: i18n.translate('expressions.functions.overallSum.args.outputColumnNameHelpText', {
+        defaultMessage: 'Name of the column to store the resulting overall sum in',
+      }),
+      types: ['string'],
+      required: false,
+    },
+  },
+
+  fn(input, { by, inputColumnId, outputColumnId, outputColumnName, metric }) {
+    const resultColumns = buildResultColumns(
+      input,
+      outputColumnId,
+      inputColumnId,
+      outputColumnName
+    );
+
+    if (!resultColumns) {
+      return input;
+    }
+
+    const accumulators: Partial<Record<string, number>> = {};
+    const valueCounter: Partial<Record<string, number>> = {};
+    input.rows.forEach((row) => {
+      const bucketIdentifier = getBucketIdentifier(row, by);
+      const accumulatorValue = accumulators[bucketIdentifier] ?? 0;
+
+      const currentValue = row[inputColumnId];
+      if (currentValue != null) {
+        switch (metric) {
+          case 'average':
+            valueCounter[bucketIdentifier] = (valueCounter[bucketIdentifier] ?? 0) + 1;
+          case 'sum':
+            accumulators[bucketIdentifier] = accumulatorValue + Number(currentValue);
```

Comment:
It looks like you're handling array-value fields in `average` but not in `sum`, is that right? Will this fail for array values, or `null`?